test/server/cards/06-CotE/MiyakosUndertaking.spec.js

```diff
@@ -0,0 +1,175 @@
+                this.player1.clickCard('miyako-s-undertaking');
+                this.player1.clickCard(this.akodoGunso);
+                this.player1.clickCard(this.bayushiLiar);
+                let handSize = this.player1.player.hand.size();
+                this.player2.clickCard('assassination');
+                this.player2.clickCard(this.bayushiLiar);
+                expect(this.player1.player.hand.size()).toBe(handSize);
+            });
+
+            it('should add any relevant keywords', function() {
+                this.player1.clickCard('miyako-s-undertaking');
+                this.player1.clickCard(this.akodoGunso);
+                this.player1.clickCard(this.bayushiLiar);
+                this.noMoreActions();
+                expect(this.player1).toHavePrompt('Any interrupts to Akodo Gunsō being honored?');
+            });
+
+            it('should allow dashes to be changed', function() {
+                this.player1.clickCard('miyako-s-undertaking');
+                this.player1.clickCard(this.akodoGunso);
+                this.player1.clickCard(this.bayushiLiar);
+                expect(this.bayushiLiar.militarySkill).toBe(2);
+            });
+
+            it('should correctly copy the most recent effect', function() {
+                this.player1.clickCard('miyako-s-undertaking');
+                this.player1.clickCard(this.kitsuSpiritcaller);
+                this.player1.clickCard(this.bayushiKachiko);
+                this.player2.pass();
+                this.player1.clickCard('miyako-s-undertaking', 'hand');
+                this.player1.clickCard(this.implacableMagistrate);
+                this.player1.clickCard(this.bayushiKachiko);
+                this.player2.pass();
+                this.player1.clickCard(this.bayushiKachiko);
+                expect(this.player1).toHavePrompt('Conflict Action Window');
+                expect(this.bayushiKachiko.name).toBe(this.implacableMagistrate.name);
+                expect(this.bayushiKachiko.getCost()).toBe(this.implacableMagistrate.getCost());
+                expect(this.bayushiKachiko.getBaseMilitarySkill()).toBe(this.implacableMagistrate.printedMilitarySkill);
+                expect(this.bayushiKachiko.getPoliticalSkill()).toBe(this.implacableMagistrate.printedPoliticalSkill);
+                expect(this.bayushiKachiko.getTraits()).not.toContain('courtier');
+                expect(this.bayushiKachiko.getTraits()).not.toContain('shugenja');
+                expect(this.bayushiKachiko.getTraits()).not.toContain('water');
+                expect(this.bayushiKachiko.getTraits()).toContain('bushi');
+                expect(this.bayushiKachiko.getTraits()).toContain('imperial');
+                expect(this.bayushiKachiko.getTraits()).toContain('magistrate');
+                expect(this.bayushiKachiko.isUnique()).toBe(true);
+                expect(this.game.currentConflict.attackerSkill).toBe(2);
+            });
+        });
```

Comment:
Probably want a test to check that the character reverts back after the conflict